yarl/_url.py

```diff
@@ -1309,8 +1320,7 @@ def _get_str_query(self, *args: Any, **kwargs: Any) -> Union[str, None]:
         if query is None:
             return None
         if isinstance(query, Mapping):
```

Comment:
This happens for `MultiDict` and `CIMultiDict` but there really shouldn't be tuple/list values in multi-dicts

Reply:
We likely shouldn't have `tuple` and `list` values in a `MultiDict` and the comments imply its not allowed, but the code didn't prevent it so we end up with the slower path for the common case.  It would be a breaking change to alter it now.